File: python/rentable.py
from datetime import datetime, timedelta

class Rentable:
    def __init__(self, name):
        self.name = name
        self.rented = False

    @property
    def name(self):
        return self.__name
    
    @name.setter
    def name(self, value):
        if(value == ""):
            raise ValueError("Please, enter a name");
        else:
            self.__name = value

    @property
    def rented(self):
        return self.__rented
    
    @rented.setter
    def rented(self, value):
        self.__rented = value
    
    def countDays(self, duration):
        today = datetime.now()
        newDate = today + timedelta(days=duration)
        return newDate.strftime("%Y-%m-%d")
    
    def rent(self, duration):
        if(duration <= 0):
            raise ValueError("Please, enter a positive duration")
        else:
            if(self.rented == False):
                self.rented = True
                print(f"You rented a {self.name} for {duration} days")
            else:
                print(f"This item is already rented")
        availableDate = self.countDays(duration)
        print((f"The rental will be terminated until {availableDate}"))

File: java/src/Rentable.java
import java.time.LocalDate;
import java.time.LocalDateTime;

public class Rentable {
    public String name;
    private Boolean rented = false;

    public Rentable(String name) {
        this.name = name;
    }

    public String getName() {
        return name;
    }

    public Boolean getRented() {
        return rented;
    }

    public void setName(String name) {
        if(name == null || name.isEmpty()){
            throw new  IllegalArgumentException("Please, enter a name");
        }
        this.name = name;
    }

    public void rent(float duration) {
        if(duration <= 0) {
            throw  new IllegalArgumentException("Please, enter a positive duration");
        }
        else {
            if (this.rented == false) {
                this.rented = true;
                System.out.println("You rented a " + this.name + " for " + duration + " days.");
            } else {
                System.out.println("This item is already rented");
            }
        }
        LocalDate availableDate = countDays(duration);
        System.out.println("The rental will be terminated until "+availableDate);
    }

    private LocalDate countDays(float duration) {
        LocalDateTime now = LocalDateTime.now();
        int aDayToHours = 24;
        int anHourToMinutes = 60;
        int aMinuteToSeconds = 60;
        LocalDateTime newDate = now.plusSeconds((long)(duration*aDayToHours*anHourToMinutes*aMinuteToSeconds));
        return newDate.toLocalDate();
    }

    @Override
    public String toString() {
        return "Rentable{" +
                "name='" + name + '\'' +
                ", rented=" + rented +
                '}';
    }
}
